src/main/java/pl/magisterka/dominikszojda/repository/DogOwnerRepository.java

```diff
@@ -0,0 +1,18 @@
+package pl.magisterka.dominikszojda.repository;
+
+
+import org.springframework.data.repository.CrudRepository;
+import org.springframework.stereotype.Repository;
+import pl.magisterka.dominikszojda.entity.DogOwnerEntity;
+
+import java.util.UUID;
+
+@Repository
+public interface DogOwnerRepository  extends CrudRepository<DogOwnerEntity, UUID> {
+
+    public DogOwnerEntity findDogOwnerById(UUID id);
+
+    public DogOwnerEntity findDogOwnerByName(String name);
```

Comment:
nazwy można (ale nie trzeba) uprościć do `findById` (i `findByName`) ponieważ już samego `extends CrudRepository<`**`DogOwnerEntity`**`, UUID>` wynika po jakiej tabelce szukasz oraz z typu return wynika co szukasz. 

W interfejsach wszystko by default jest `public` więc nie zwykło się pisać modyfikatora `public`. 

Dobrym pomysłem jest zwracać `Optional<TwójTyp>` zamiast `TwójTyp` ponieważ zapewnia to null safety. Zwracając `Optional<DogOwnerEntity>` z repo masz gwarancję że nigdy nie dostaniesz null i można łatwo napisać kod, który jest prosty i przejrzysty, np.
```java
// be optional
Entity e = repo.find();
if (e == null) throw new EntityNotFoundException();
return e;

// z Optional:
return repo.find().orElseThrow(EntityNotFoundException::new);
```
Kod się dodatkowo może uprościć jeżeli potrzebne jest jakieś dodatkowe mapowanie na jakiś response czy coś. 

```suggestion
    Optional<DogOwnerEntity> findById(UUID id);

    Optional<DogOwnerEntity> findByName(String name);
```